### PR TITLE
Reduce the use of `!important` for GitHub banner

### PR DESCRIPTION
**Pull request by stevenjoezhang**
## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes was maked (for bug fixes / features).
   - [x] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs in [NexT website](https://theme-next.org/docs/) have been added / updated (for new features).

## PR Type
**What kind of change does this PR introduce?**

- [ ] Bugfix.
- [ ] Feature.
- [x] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [ ] Other... Please describe:

## What is the current behavior?

Issue resolved: N/A

## What is the new behavior?

- Screenshots with this changes: N/A
- Link to demo site with this changes: N/A

### How to use?
In NexT `_config.yml`:
```yml
...
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

@wip ready for review

### CODE REVIEW

**Comment by stevenjoezhang** [comment]
@ivan-nginx The title of this PR is called **`Reduce the use of !important`**

File: source/css/_common/components/header/github-banner.styl [hunk 1]
```diff
@@ -6,22 +18,19 @@
...
+    color: #fff;
+    position: absolute;
+    top: 0;
+    border: 0;
+    right: 0;
   }
   +tablet-mobile() {
     > svg {
       if (scheme == 'Pisces') || (scheme == 'Gemini') {
-        fill: #fff !important;
```
Comment (stevenjoezhang) [inline source/css/_common/components/header/github-banner.styl]: Here. Please, have a look.

Comment (ivan-nginx) [reply source/css/_common/components/header/github-banner.styl]: U say u fix `hide()` with `!important`. GitHub banner nothing to do with this.

Comment (stevenjoezhang) [reply source/css/_common/components/header/github-banner.styl]: My mistake, it's not just about `hide()`
And I have tested, it's fine in all resolution, doesn't brake foreground styles

**Comment by stevenjoezhang** [comment]
I believe this PR is ready to be merged now. No other work needs to be done

File: source/css/_schemes/Gemini/index.styl [hunk 2]
```diff
@@ -76,8 +76,7 @@ use_seo = hexo-config('seo');
 }
 
 // Post delimiters.
-.post-eof,
-.post-spread {
```
Comment (ivan-nginx) [inline source/css/_schemes/Gemini/index.styl]: Are we remove this class? I don't remember...

Comment (stevenjoezhang) [reply source/css/_schemes/Gemini/index.styl]: I didn't find `post-spread` in NexT... seems it has been removed

File: source/css/_common/components/header/github-banner.styl [hunk 3]
```diff
@@ -34,8 +43,7 @@
   +mobile() {
     > svg {
       if (scheme == 'Mist') {
-        top: inherit !important;
-        margin-top: -50px;
```
Comment (ivan-nginx) [inline source/css/_common/components/header/github-banner.styl]: I believe u not test on all schemes with all resolutions:

image

Don't need remove parameters if u don't know for what they are. If parameter present, then he's needed for something.

File: source/css/_common/components/post/post-eof.styl [hunk 4]
```diff
@@ -10,7 +9,7 @@
 }
 
 .post:last-child {
-  .post-eof.post-eof.post-eof {
```
Comment (ivan-nginx) [inline source/css/_common/components/post/post-eof.styl]: Screen? Where is this class?

File: source/css/_schemes/Gemini/index.styl (hunk 2 above)
Comment (stevenjoezhang) [reply source/css/_schemes/Gemini/index.styl]: Here: https://github.com/theme-next/hexo-theme-next/pull/357/files. `.post-spread` no longer exist

File: source/css/_common/components/post/post-eof.styl (hunk 4 above)
Comment (stevenjoezhang) [reply source/css/_common/components/post/post-eof.styl]: Not sure. But it same as `.post-eof`, no need to repeat 3 times

File: source/css/_common/components/header/github-banner.styl (hunk 3 above)
Comment (stevenjoezhang) [reply source/css/_common/components/header/github-banner.styl]: Of course I know what I'm doing. It looks fine in my browser. 
屏幕快照 2019-03-30 上午1 56 17
Are you using the latest branch?

File: source/css/_common/components/post/post-eof.styl (hunk 4 above)
Comment (ivan-nginx) [reply source/css/_common/components/post/post-eof.styl]: Oh, my fault: i forgot to checkout your pull.

File: source/css/_common/components/header/github-banner.styl (hunk 3 above)
Comment (ivan-nginx) [reply source/css/_common/components/header/github-banner.styl]: Oh, my fault: i forgot to checkout your pull.

File: source/css/_common/components/post/post-eof.styl (hunk 4 above)
Comment (ivan-nginx) [reply source/css/_common/components/post/post-eof.styl]: This thing need to add delimeter between posts in homepage. And of course, last delimeter no needed, that's why display `none` on last child.

Comment (ivan-nginx) [reply source/css/_common/components/post/post-eof.styl]: But see, in Gemini I make same feature with another way:

https://github.com/theme-next/hexo-theme-next/blob/022b9ccb0ef1086b06039737e150dc27faae9e2c/source/css/_schemes/Gemini/index.styl#L45-L53

IDK which method is better, but I think need to choose only one.

Comment (ivan-nginx) [reply source/css/_common/components/post/post-eof.styl]: So, in Muse and Pisces – `.post-eof` enabled.
In Mist – disabled and in Gemini – disabled & implemented with another way.

**Comment by 1v9** [comment]
Sorry @stevenjoezhang I made test again for WIP, there's a pending status of WIP, add `@wip ready for review` in PR body can pass pending, look like this: `WIP Successful in 1s — Ready for review`

PR template need to update 😉

**Comment by stevenjoezhang** [comment]
Anything else need to do with this PR?

**Pull request merged**